rest/admin_api_test.go

```diff
@@ -2136,3 +2136,42 @@ func TestUserAndRoleResponseContentType(t *testing.T) {
 	assert.Equal(t, "application/json", response.Header().Get("Content-Type"))
 	require.NoError(t, base.JSONUnmarshal(response.Body.Bytes(), &responseBody))
 }
+
+func TestConfigRedaction(t *testing.T) {
+	rt := NewRestTester(t, &RestTesterConfig{DatabaseConfig: &DbConfig{Users: map[string]*db.PrincipalConfig{"alice": {Password: base.StringPtr("password")}}}})
+	defer rt.Close()
+
+	// Test default db config redaction
+	var unmarshaledConfig DbConfig
+	response := rt.SendAdminRequest("GET", "/db/_config", "")
+	err := json.Unmarshal(response.BodyBytes(), &unmarshaledConfig)
+	require.NoError(t, err)
+
+	assert.Equal(t, "****", unmarshaledConfig.Password)
+	assert.Equal(t, "****", *unmarshaledConfig.Users["alice"].Password)
+
+	// Test default db config redaction when redaction disabled
+	response = rt.SendAdminRequest("GET", "/db/_config?redact=false", "")
+	err = json.Unmarshal(response.BodyBytes(), &unmarshaledConfig)
+	require.NoError(t, err)
+
+	assert.Equal(t, "password", unmarshaledConfig.Password)
```

Comment:
Here (and below in the serverConfig response), we might need to be set to use `base.TestClusterPassword()` in order to allow it to pass on an integration test that has a non-default password (via `SG_TEST_PASSWORD`)

Reply:
Ah cool. Didn't know that was a thing. Will make that change